compose/config/config.py

```diff
@@ -199,14 +238,52 @@ def load(config_details):
 
     Return a fully interpolated, extended and validated configuration.
     """
+    version = get_config_version(config_details)
+    if version not in COMPOSEFILE_VERSIONS:
+        raise ConfigurationError('Invalid config version provided: {0}'.format(version))
```

Comment:
I still think this is an issue, but not enough to block this branch, so I guess we can discuss it after this is merged.